### PR TITLE
Add staging environment

### PR DESCRIPTION
**Pull request by enricostano**
- [x] Add staging to `config/secrets.yml`

### CODE REVIEW

File: config/environments/staging.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  # Send deprecation notices to registered listeners.
+  config.active_support.deprecation = :notify
+
+  # Use default logging formatter so that PID and timestamp are not suppressed.
+  config.log_formatter = ::Logger::Formatter.new
+
+  # Do not dump schema after migrations.
+  config.active_record.dump_schema_after_migration = false
+end
```
Comment (sauloperez) [inline config/environments/staging.rb]: is this a copy of production's config?

Comment (enricostano) [reply config/environments/staging.rb]: Yep, we need to change the mail sender though

File: config/environments/staging.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  config.action_mailer.default_url_options = {
+    host: ENV["MAIL_LINK_HOST"],
+    protocol: (ENV["MAIL_LINK_PROTO"] || "https")
+  }
+
+  smtp_env = Hash[ENV.map do |k,v|
+    if /^SMTP_(.*)$/ === k
+      [$1.downcase.to_sym, YAML.load(v)]
+    end
+  end.compact]
```
Comment (enricostano) [inline config/environments/staging.rb]: Understand this, document it and add staging stuff.

Comment (enricostano) [reply config/environments/staging.rb]: done

File: config/environments/staging.rb (hunk 1 above)
Comment (enricostano) [reply config/environments/staging.rb]: Will do in another PR

**Pull request merged**